test/unit/math/opencl/prim/cholesky_decompose_test.cpp

```diff
@@ -0,0 +1,99 @@
+#ifdef STAN_OPENCL
+#include <stan/math/prim/mat.hpp>
+#include <stan/math/opencl/opencl.hpp>
+#include <gtest/gtest.h>
+#include <algorithm>
+#include <vector>
+
+#define EXPECT_MATRIX_NEAR(A, B, DELTA) \
+  for (int i = 0; i < A.size(); i++)    \
+    EXPECT_NEAR(A(i), B(i), DELTA);
+
+TEST(MathMatrix, cholesky_decompose_cl_expections) {
```

Comment:
These are the same tests as for the in-place cholesky, just on the non-inplace one.